cubical/data_handler/TiggerSourceProvider.py

```diff
@@ -96,48 +108,30 @@ def point_stokes(self, context):
         """ Returns a stokes parameter array to Montblanc. """
 
         # Get the extents of the time, baseline and chan dimension
-        (lt, ut), (lp, up) = context.dim_extents('ntime', 'npsrc')
-
+        (lp, up), (lt, ut), (lc, uc) = context.dim_extents('npsrc',
+                                                           'ntime',
+                                                           'nchan')
+        # (npsrc, ntime, nchan, 4)
         stokes = np.empty(context.shape, context.dtype)
 
-        for ind, source in enumerate(self._pnt_sources[lp:up]):
-            stokes[ind,:,0] = source.flux.I
-            stokes[ind,:,1] = source.flux.Q
-            stokes[ind,:,2] = source.flux.U
-            stokes[ind,:,3] = source.flux.V
-
-        return stokes
-
-    def point_alpha(self, context):
-        """ Returns a spectral index (alpha) array to Montblanc. """
-
-        alpha = np.empty(context.shape, context.dtype)
-
-        (lp, up) = context.dim_extents('npsrc')
+        f = self._freqs[None, lc:uc]
 
         for ind, source in enumerate(self._pnt_sources[lp:up]):
             try:
-                alpha[ind] = source.spectrum.spi
-            except:
-                alpha[ind] = 0
+                rf = source.spectrum.freq0
+            except AttributeError:
+                # TODO(jkenyon, osmirnov)
+                # I think this should be "or 1"
+                rf = self._sm.freq0 or 0
 
-        return alpha
-
-    def point_ref_freq(self, context):
-        """ Returns a reference frequency per source array to Montblanc. """
-        
-        pt_ref_freq = np.empty(context.shape, context.dtype)
-
-        (lp, up) = context.dim_extents('npsrc')
-        
-        for ind, source in enumerate(self._pnt_sources[lp:up]):
-            try:
-                pt_ref_freq[ind] = source.spectrum.freq0
-            except:
-                pt_ref_freq[ind] = self._sm.freq0 or 0
+            a = source.spectrum.spi
 
-        return pt_ref_freq
+            stokes[ind, :, :, 0] = source.flux.I[None, None]*(f/rf)**a
```

Comment:
Better to compute the spectrum ``(f/rf)**a`` just once and reuse. Especially since we're about to put in higher-order spis.